app/src/main/java/com/example/drawingApp/DialogObject.kt

```diff
@@ -0,0 +1,119 @@
+package com.example.drawingApp
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.EditText
+import android.widget.TextView
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.core.view.isVisible
+import androidx.core.widget.addTextChangedListener
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.bottomsheet.BottomSheetDialog
+
+object DialogObject {
+
+    fun showLayerAlertDialog(
+        c: Context,
+        model: DrawingViewModel,
```

Comment:
I think you should consider renaming this to viewModel, having a model in here was a little confusing compared to calling it a viewModel

Reply:
I think thats a fair point, incorporated.